lib/ContextModuleFactory.js

```diff
@@ -111,12 +109,10 @@ module.exports = class ContextModuleFactory extends ModuleFactory {
 
 				const contextResolver = this.resolverFactory.get(
 					"context",
-					resolveOptions || EMPTY_RESOLVE_OPTIONS
-				);
-				const loaderResolver = this.resolverFactory.get(
-					"loader",
-					EMPTY_RESOLVE_OPTIONS
+					resolveOptions || undefined,
+					dependencies.length > 0 ? dependencies[0].category : undefined
```

Comment:
```suggestion
					dependencies.length > 0
					 ? cachedSetProperty(resolveOptions || EMPTY_RESOLVE_OPTIONS, "dependencyType", dependencies[0].category)
					 : resolveOptions || undefined
```

You need to add a new `cacheSetProperty` next to `cachedCleverMerge` and do some WeakMap magic for caching.